x-pack/plugins/cloud/public/plugin.test.ts

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { coreMock } from 'src/core/public/mocks';
+import { homePluginMock } from 'src/plugins/home/public/mocks';
+import { securityMock } from '../../security/public/mocks';
+import { CloudPlugin } from './plugin';
+
+const nextTick = async () => await new Promise<void>((resolve) => setTimeout(() => resolve(), 1));
+
+describe('Cloud Plugin', () => {
+  describe('#start', () => {
+    function setupPlugin({ roles = [] }: { roles?: string[] } = {}) {
+      const plugin = new CloudPlugin(
+        coreMock.createPluginInitializerContext({
+          id: 'cloudId',
+          base_url: 'https://cloud.elastic.co',
+          deployment_url: '/abc123',
+          profile_url: '/profile/alice',
+          organization_url: '/org/myOrg',
+        })
+      );
+      const coreSetup = coreMock.createSetup();
+      const homeSetup = homePluginMock.createSetupContract();
+      const securitySetup = securityMock.createSetup();
+      securitySetup.authc.getCurrentUser.mockResolvedValue({
```

Comment:
__optional nit:__ wondering if it's time to expose existing `mockAuthenticatedUser` from `security/common/mocks`? Not a big deal, up to you.

Reply:
Good idea, will do! I forgot about this mock